packages/lisk-core/src/factories/components.js

```diff
@@ -0,0 +1,22 @@
+const logger = require('@lisk/lisk-logger-bunyan');
+
+const componentsList = {};
+const componentsConfigCache = {};
+
+const ComponentFactory = {
+	register: (component, constructorFunction) => {
+		componentsList[component] = constructorFunction;
+	},
+
+	create: (component, config = null) => {
+		if (config) {
+			componentsConfigCache[component] = config;
+		}
+
+		return componentsList[component](componentsConfigCache[component]);
+	},
+};
+
+ComponentFactory.register('logger', logger);
+
+module.exports = ComponentFactory;
```

Comment:
I'm against using this object since we can already do the same thing with commonjs modules.